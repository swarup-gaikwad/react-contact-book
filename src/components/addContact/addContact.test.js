import React from "react";
import { shallow } from "enzyme";
import AddContact from "./addContact";

describe("AddContact", () => {
  it("should render correctly", () => {
    const component = shallow(<AddContact />);
  });
  it("should render initial layout", () => {
    // when
    const component = shallow(<AddContact />);
    // then
    expect(component.getElements()).toMatchSnapshot();
  });
});
