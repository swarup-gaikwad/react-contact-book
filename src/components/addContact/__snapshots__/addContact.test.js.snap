// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContact should render initial layout 1`] = `
Array [
  <div
    className="container"
  >
    <h2>
      Add Contact
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="fName"
        >
          First Name
        </label>
        <input
          className="form-control"
          name="fName"
          onChange={[Function]}
          placeholder="Enter First Name"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="lName"
        >
          Last Name
        </label>
        <input
          className="form-control"
          name="lName"
          onChange={[Function]}
          placeholder="Enter Last Name"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="mobile"
        >
          Mobile Number
        </label>
        <input
          className="form-control"
          name="mobile"
          onChange={[Function]}
          pattern="[0-9]{10}"
          placeholder="Enter Mobile Number"
          title="Please enter valid 10 digit mobile number"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="empId"
        >
          Employee Id
        </label>
        <input
          className="form-control"
          name="empId"
          onChange={[Function]}
          placeholder="Enter Employee Id"
          required={true}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Add Contact
      </button>
    </form>
  </div>,
]
`;
